crypto/rsa/rsa_impl.c

```diff
@@ -526,6 +526,8 @@ static int rsa_private_transform(RSA *rsa, uint8_t *out, const uint8_t *in,
     goto err;
   }
 
+  BN_set_flags(f, BN_FLG_CONSTTIME);
+
```

Comment:
OK, so IIRC it is because you added an `assert(BN_get_flags(I, BN_FLG_CONSTTIME));`, right? And, without this `BN_set_flags`, that fails, right?

Check this out:

```
#define BN_mod(rem, numerator, divisor, ctx) \
  BN_div(NULL, (rem), (numerator), (divisor), (ctx))
```
In mod_exp:
```
BN_with_flags(c, I, BN_FLG_CONSTTIME);
BN_mod(r1, c, rsa->q, ctx));
```
the `BN_mod` expands to:
```
BN_div(NULL, r, c, rsa->q, (ctx))
```

So, `c` is `BN_div`'s `num` parameter and `rsa->q` is `BN_div`'s `divisor` parameter.

Let's look at `BN_div`:
```
int BN_div(BIGNUM *dv, BIGNUM *rm, const BIGNUM *num, const BIGNUM *divisor,
           BN_CTX *ctx) {
  [...]
  if ((num->flags & BN_FLG_CONSTTIME) != 0 ||
      (divisor->flags & BN_FLG_CONSTTIME) != 0) {
    no_branch = 1;
  }
```

So, instead of setting the BN_FLG_CONSTTIME flag here, you can `assert(BN_get_flags(rsa->q, BN_FLG_CONSTTIME))` instead of asserting the same for `I`. Then you won't need this.